src/box/lua/net_box.lua

```diff
@@ -243,12 +245,17 @@ local function create_transport(host, port, user, password, callback,
     ::stop::
             send_buf:recycle()
             recv_buf:recycle()
+            on_send_buf_empty:broadcast()
             worker_fiber = nil
         end)
     end
 
     local function stop()
         if not is_final_state[state] then
+            is_closing = true
+            while (send_buf:size() ~= 0) do
+                on_send_buf_empty:wait()
+            end
```

Comment:
Let's add a comment here: what's going on, why this loop will eventually break (because all new requests will fail because of is_closing).

Reply:
fixed